### PR TITLE
fix(DB/gameobject_loot_template): Correct drop chances for all classic ores

### PR DESCRIPTION
**Pull request by Hacki95**
Co-Author: @necropola

## Changes Proposed:
- Correct the dropchances of all types of "Stones" from classic ores.
- Correct the dropchances of all types of "Ores" from classic ores.
- Correct some groupIds

## Issues Addressed:
- Closes: https://github.com/chromiecraft/chromiecraft/issues/448
- Closes: https://github.com/azerothcore/azerothcore-wotlk/issues/5625

## SOURCE:

- I used a mix of classic wowhead data, math and simple guessing. Since wotlk nodes aren't looted 3 times anymore but only once. So for most numbers I simply took the min/max value and multiplied it by 3. Since some drops already had 2-4 instead of 1-3 I went with similiar numbers for the others.
- The numbers might be guessed but they are way closer to the correct ones I assume. For example most stone types had a drop chance of 100%. Every source points to them being a more moderate 40% so I went with that number.
- I am not 100% sure every min/max number is completely correct now but they should only be off by 1 if they are. Not much of a chance to be 100% sure until wotlk classic hits. Its way better than before atleast.

## Tests Performed:

- Applied the SQL
- Checked the Database

## How to Test the Changes:

- Apply the SQL
- Go do some mining

## Known Issues and TODO List:

- No way to be 100% sure about min/max values atm but they are way closer to the correct ones than before. If it turns out they are off by about 1 its a simple fix.

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1620115611701757811.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
+-- INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1620115611701757811');
+
+-- Copper Vein
+UPDATE `gameobject_loot_template` SET `MaxCount`=7 WHERE FIND_IN_SET (`Entry`,'1502,1735,2626,18092') and Item = 2835; -- Rough Stone
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1620115611701757811.sql:4]: Big AND

Comment (Hacki95) [reply data/sql/updates/pending_db_world/rev_1620115611701757811.sql]: Fixed

**Review by FrancescoBorzi — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1620115611701757811.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
+-- INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1620115611701757811');
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1620115611701757811.sql:1]: why commented? we need it

Comment (Hacki95) [reply data/sql/updates/pending_db_world/rev_1620115611701757811.sql]: Ohh you are right I always comment this out while testing. Fixed!

**Comment by necropola** [comment]
**An overview of the expected effect on the average drop rate for the different types of Ore & Stone**

image

**Comment by Hacki95** [comment]
Found some better data by using the waybackmachine on wowhead and analyzing the source from the site.

I updated the google sheet. Huge thanks again to @necropola for making it.
grafik

File: data/sql/updates/pending_db_world/rev_1620115611701757811.sql [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@HEAVYSTONE = 2838,
+@IRONORE = 2772,
+@SOLIDSTONE = 7912,
+@MITHRILORE = 3858,
+@DENSESTONE = 12365,
+@SOULDARITE = 19774,
+@THORIUMORE = 10620;
+
+-- Copper Vein
+UPDATE `gameobject_loot_template` SET `MaxCount`=6 WHERE FIND_IN_SET (`Entry`,'1502,1735,2626,18092') AND Item = @ROUGHSTONE; -- Rough Stone
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1620115611701757811.sql:17]: backtick on item

**Comment by necropola** [comment]
I've added an additional column to the sheet which contains the `Loot entry (Data1)` value and I've grouped the rows by this value. This makes it a **lot** easier to match the table with your PR.

image

File: data/sql/updates/pending_db_world/rev_1620115611701757811.sql [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+-- Tin Vein
+UPDATE `gameobject_loot_template` SET `MaxCount`=6 WHERE FIND_IN_SET (`Entry`,'1503,1736,2627,18093') AND `Item` = @COARSESTONE; -- Coarse Stone
+UPDATE `gameobject_loot_template` SET `MinCount`=1 WHERE FIND_IN_SET (`Entry`,'1503,1736,2627,18093') AND `Item` = @COARSESTONE; -- Coarse Stone
+UPDATE `gameobject_loot_template` SET `Chance`=50 WHERE FIND_IN_SET (`Entry`,'1503,1736,2627,18093') AND `Item` = @COARSESTONE; -- Coarse Stone
+UPDATE `gameobject_loot_template` SET `MinCount`=1 WHERE FIND_IN_SET (`Entry`,'1503,1736,2627,18093') AND `Item` = @TINORE; -- Tin Ore
+UPDATE `gameobject_loot_template` SET `MaxCount`=4 WHERE FIND_IN_SET (`Entry`,'1503,1736,2627,18093') AND `Item` = @TINORE; -- Tin Ore
+
+-- Iron Deposit
+UPDATE `gameobject_loot_template` SET `Chance`=50 WHERE Entry = 1505 AND `Item` = @HEAVYSTONE; -- Heavy Stone
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1620115611701757811.sql:31]: backtick on entry too, all keywords should have them =)

Comment (necropola) [reply data/sql/updates/pending_db_world/rev_1620115611701757811.sql:31]: You mean column names, right?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1620115611701757811.sql:31]: yes

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1620115611701757811.sql:31]: https://www.azerothcore.org/wiki/sql-standards

**Pull request merged**